test/hdt-test.js

```diff
@@ -38,6 +38,58 @@ describe('hdt', function () {
     });
   });
 
+  describe('modifying an HDT header', function () {
+    var document;
+    before(function () {
+      // We're modifying the hdt, so copy it to a different location first
+      return new Promise((resolve, reject) => {
+        fs.createReadStream('./test/test.hdt')
+          .on('error', reject)
+          .pipe(fs.createWriteStream('./test/.tmp/test.hdt'))
+          .on('error', reject)
+          .on('finish', resolve);
+      })
+        .then(() => hdt.fromFile('./test/.tmp/test.hdt'))
+        .then(hdtDocument => {
+          document = hdtDocument;
+        });
+    });
+
+    after(function () {
+      return document.close();
+    });
+
+    describe('reading, modifying and writing a new header', function () {
+      var oldHeader, newHeader;
+      before(function () {
+        return document.readHeader()
+          .then(result => {
+            oldHeader = result.slice();
+            result.splice(0, 1);
```

Comment:
Can we have a more atomic test where we write an array of a single triple or something? It's very hard to see what's going on here, plus we also depend on `readHeader`.